src/main/java/org/folio/rest/impl/ModTenantAPI.java

```diff
@@ -46,10 +46,14 @@ public ModTenantAPI() {
 
   @Override
   Future<Integer> loadData(TenantAttributes attributes, String tenantId, Map<String, String> headers, Context context) {
-    if (fileSplittingEnabled) {
-      systemUserAuthService.initializeSystemUser(headers);
-    }
-    return super.loadData(attributes, tenantId, headers, context)
+    return Future.succeededFuture()
+      .compose(v -> {
+        if (fileSplittingEnabled) {
+          return systemUserAuthService.initializeSystemUser(headers);
```

Comment:
I think the system user should be created everytime. If a tenant is installed with file splitting disabled, it cannot have DI working when file splitting is enabled.

Reply:
Our goal was to not have file splitting be a breaking change/only opt-in; IMO having to set its password environment variable is a breaking change.

I'll add a note to the release notes though; in generally, I'd advise calling install whenever changing configuration like this.  The system user's password env variable would have the same issue, where an install is required each time

Reply:
Also, I don't want to spook admins with a new user appearing out of the blue

Reply:
Sounds good!